INCWaveMountainsView/Classes/INCWaveMountainsView.m

```diff
@@ -0,0 +1,591 @@
+
+-(void)_removeMountainPosition:(INCMountainPosition)mountainPosition
+{
+    switch (mountainPosition) {
+        case INCMountainPositionLeft:
+            if (self.leftMountainPointId) {
+                [self.percentReachFull setObject:@YES forKey:self.leftMountainPointId];
+            }
+            self.leftMountainPercent = -1;
+            self.leftMountainPointId = NULL;
+            break;
+        
+        case INCMountainPositionCenter:
+            if (self.centerMountainPointId) {
+                [self.percentReachFull setObject:@YES forKey:@(self.centerMountainPercent)];
+            }
+            self.centerMountainPercent = -1;
+            self.centerMountainPointId = NULL;
+            break;
+            
+        case INCMountainPositionRight:
+            if (self.rightMountainPointId) {
+                [self.percentReachFull setObject:@YES forKey:@(self.rightMountainPercent)];
+            }
+            self.rightMountainPercent = -1;
+            self.rightMountainPointId = NULL;
+            
+            break;
+    }
+}
+
+#pragma mark - Drawing functions
+
+static float distanceBeetweenBackgroundLines = 10;
+-(void)_drawBackgroundLinesInRect:(CGRect)rect withContext:(CGContextRef)context
+{
+    int numberOfLines = (rect.size.width / 10) - 1;//We substract one line because we don't want the line at the end
+    
+    CGContextSetStrokeColorWithColor(context, [UIColor colorWithRed:151.0/255.0 green:151.0/255.0 blue:151.0/255.0 alpha:0.11].CGColor);
+    CGContextSetLineWidth(context, 1.0f);
+    
+    for (int i = 0; i < numberOfLines; i++) {
+        
+        float yPosLine = (float)(i + 1) * distanceBeetweenBackgroundLines;
+        
+        CGContextMoveToPoint(context, 0.0f, yPosLine); //start at this point
+        
+        CGContextAddLineToPoint(context, rect.size.width, yPosLine); //draw to this point
+        
+        // and now draw the Path!
+    }
+    CGContextStrokePath(context);
+
+}
+
+- (UIBezierPath *)_pathWithIdPoint:(NSInteger)idPoint percent:(float)percent inMountain:(INCMountainPosition)mountain
+{
+    int index = -1;
+    
+    CAShapeLayer *layer;
+    switch (mountain) {
+        case INCMountainPositionLeft:
+            
+            layer = self.shapeLayerLeftMountain;
+            self.leftMountainPercent = percent;
+            self.leftMountainPointId = @(idPoint);
+            
+            index = 0;
+            break;
+            
+        case INCMountainPositionCenter:
+            
+            layer = self.shapeLayerCenterMountain;
+            self.centerMountainPercent = percent;
+            self.centerMountainPointId = @(idPoint);
+            
+            index = 1;
+            break;
+            
+        case INCMountainPositionRight:
+            
+            layer = self.shapeLayerRightMountain;
+            self.rightMountainPercent = percent;
+            self.rightMountainPointId = @(idPoint);
+            index = 2;
+            
+            break;
```

Comment:
Mmm maybe I can do it, but maybe it's more clear what index belongs to each type? What do you think? The code repeated now is not too much